src/components/Listing/Listing.jsx

```diff
@@ -0,0 +1,65 @@
+import DisplayCheckboxGroup from '../common/DisplayCheckboxGroup';
+import DisplayInput from '../common/DisplayInput';
+
+/**
+ * Represents details of a listing.
+ * @typedef {Object} ListingDetails
+ * @property {string} isbn - The ISBN of the book.
+ * @property {string} title - The title of the book.
+ * @property {number} edition - The edition of the book.
+ * @property {string} department - The department the book belongs to.
+ * @property {string} courseNumber - The course number associated with the book.
+ * @property {number} price - The price of the book.
+ * @property {User} buyer - The buyer of the book.
+ * @property {User} seller - The seller of the book.
+ */
+
+/**
+ * Represents a user.
+ * @typedef {Object} User
+ * @property {string} id - The ID of the user.
+ * @property {string} email - The email address of the user.
+ * @property {string} preferredContactMethod - The preferred contact method of the user.
+ * @property {string[]} acceptedPaymentMethods - The accepted payment methods by the user.
+ */
+
+/**
+ * Functional component for displaying a listing.
+ * @param {Object} props - The component props.
+ * @param {ListingDetails} props.listingData - The data representing the listing.
+ * @returns {JSX.Element} JSX element representing the listing.
+ */
```

Comment:
Are we expecting this from all new components? To make sure everyone does this?

Reply:
No just something Im trying out.
This is JSDocs which gives IDE autocomplete and type-checking which is the main benefit of TS imo